src/controller/actions/topic.js

```diff
@@ -0,0 +1,79 @@
+import api from '../../controller/api';
+
+export const TOPIC_SUBMIT = 'TOPIC_SUBMIT';
+export const TOPIC_SUCCESS = 'TOPIC_SUCCESS';
+export const TOPIC_ERROR = 'TOPIC_ERROR';
+
+export const TOPIC_DELETE = 'TOPIC_DELETE';
+
+/**
+ * Components can consume this function to get all Topics
+ * On submit: state.topic.loading === true
+ * On success: state.topic.topics === [of topics]
+ * On fail: state.topic.error === some error object
+ */
+export const getTopics = () => dispatch => {
+  dispatch(topicSubmit());
+  api.topics
+    .get()
+    .then(topics => dispatch(topicSuccess(topics)))
+    .catch(err => dispatch(topicError(err)));
+};
+
+/**
+ * Components can consume this function to add a new Topic
+ * On submit: state.topic.loading === true
+ * On success: state.topic.topics === [of topics]
+ * On fail: state.topic.error === some error object
+ * @param {{title: string}} object
+ */
+//FIXME: Line 39 I am calling getTopics again in order to refresh topics on dashboard
+// tried calling this inside of a componentDidUpdate within the AllTopicsContainer and it triggered some
+// weird behavior (calling GET non-stop). Need to rethink how this is working - clearly something
+// wrong with how it's written
+export const addTopic = title => dispatch => {
+  dispatch(topicSubmit());
```

Comment:
Right, well I see you're managing the topics in Redux state, so let's start updating the titles using state, as well.

Recall that the POST request will ultimately return the new topic object, so we can use this to update our local state without fetching the topics all over again.

We can either do this optimistically, by updating the state before the API call comes back, or pessimistically, by updating the state with the response when the promise resolves successfully. For now, let's think about doing it pessimistically since this is the most straightforward way to handle async state updates.

Perhaps you could change line 39 and the corresponding action to be:

```js
39  .then(topic => dispatch(topicSuccess(topic)))
```
...and update the reducer to push this new topic to state on TOPIC_SUCCESS actions.

Reply:
awesome. thank you!